Source/resizeablerectitem.cpp

```diff
@@ -115,34 +115,22 @@ void ResizeableRectItem::paint( QPainter* aPainter, const QStyleOptionGraphicsIt
     if (m_Pixmap.isNull())
     {
         const auto objectName = mMapObject->mObjectStructureType.c_str();
-        for (;;)
+        for (int sizeCandidate = 8; sizeCandidate > 1; sizeCandidate--)
         {
             QFont f = aPainter->font();
             QFontMetricsF fm(f);
-            QRectF textRect = fm.boundingRect(cRect, Qt::AlignCenter | Qt::TextWrapAnywhere, objectName);
-            auto size = f.pointSize();
-            if (size <= 0)
-            {
-                size = 8;
-            }
+            const auto textRect = fm.boundingRect(cRect, Qt::AlignCenter | Qt::TextWrapAnywhere, objectName);
 
             if (textRect.width() > cRect.width() ||
                 textRect.height() > cRect.height())
             {
-                if (size > 1)
-                {
-                    f.setPointSize(--size);
-                }
-                else
-                {
-                    break;
-                }
+               f.setPointSize(sizeCandidate);
```

Comment:
not sure if I'm missing something here - but shouldn't this line and aPainter->setFont(f); be before the if statement? Otherwise you are saying yes sizeCandidate fits, but its before you've set it so its actually the previous sizeCandidate or something else on the first iteration?

Reply:
or I guess they could go after so its set for the next try, hmm

Reply:
hmm.. the longer i go the harder it gets to look at this xD

```c++
      for (int sizeCandidate = 8; sizeCandidate > 1; sizeCandidate--)
      {
          QFont f = aPainter->font();
          f.setPointSize(sizeCandidate);
          aPainter->setFont(f);
          QFontMetricsF fm(f);
          const auto textRect = fm.boundingRect(cRect, Qt::AlignCenter | Qt::TextWrapAnywhere, objectName);

          if (textRect.width() < cRect.width() &&
              textRect.height() < cRect.height())
          {
              break;
          }
      }
``` whats your opinion on this approach?

Reply:
seems better if 8 is a better starting point than whatever it is when its not been set ;)